protostar-rust/src/lib.rs

```diff
@@ -317,4 +383,224 @@ version = \"0.1.0\"";
 
         assert!(err.to_string().contains("Failed to read directory at path"));
     }
+
+    #[test]
+    fn passing_none_for_filter_doesnt_filter_tests() {
+        let mocked_tests: Vec<TestConfig> = vec![
+            TestConfig {
+                name: "crate1::do_thing".to_string(),
+                available_gas: None,
+            },
+            TestConfig {
+                name: "crate2::run_other_thing".to_string(),
+                available_gas: None,
+            },
+            TestConfig {
+                name: "outer::crate2::run_other_thing".to_string(),
+                available_gas: None,
+            },
+        ];
+
+        let filtered = filter_tests_by_name(None, false, mocked_tests.clone()).unwrap();
+        assert_eq!(filtered, mocked_tests);
+    }
+
+    #[test]
+    fn filtering_tests() {
+        let mocked_tests: Vec<TestConfig> = vec![
+            TestConfig {
+                name: "crate1::do_thing".to_string(),
+                available_gas: None,
+            },
+            TestConfig {
+                name: "crate2::run_other_thing".to_string(),
+                available_gas: None,
+            },
+            TestConfig {
+                name: "outer::crate2::execute_next_thing".to_string(),
+                available_gas: None,
+            },
+        ];
+
+        let filtered = filter_tests_by_name(Some("do"), false, mocked_tests.clone()).unwrap();
+        assert_eq!(
+            filtered,
+            vec![TestConfig {
+                name: "crate1::do_thing".to_string(),
+                available_gas: None,
+            },]
+        );
+
+        let filtered = filter_tests_by_name(Some("run"), false, mocked_tests.clone()).unwrap();
+        assert_eq!(
+            filtered,
+            vec![TestConfig {
+                name: "crate2::run_other_thing".to_string(),
+                available_gas: None,
+            },]
+        );
+
+        let filtered = filter_tests_by_name(Some("thing"), false, mocked_tests.clone()).unwrap();
+        assert_eq!(
+            filtered,
+            vec![
+                TestConfig {
+                    name: "crate1::do_thing".to_string(),
+                    available_gas: None,
+                },
+                TestConfig {
+                    name: "crate2::run_other_thing".to_string(),
+                    available_gas: None,
+                },
+                TestConfig {
+                    name: "outer::crate2::execute_next_thing".to_string(),
+                    available_gas: None,
+                },
+            ]
+        );
+
+        let filtered =
+            filter_tests_by_name(Some("nonexistent"), false, mocked_tests.clone()).unwrap();
+        assert_eq!(filtered, vec![]);
+
+        let filtered = filter_tests_by_name(Some(""), false, mocked_tests).unwrap();
```

Comment:
I don't think this test is useful, there will never be a case when we get an empty string from the CLI (we will just get `None`).

Reply:
Correct, but we're also exposing this code as library and having more throughout tests usually doesn't hurt. This is a valid argument this method could take.